content/named-tuples.md

```diff
@@ -0,0 +1,777 @@
+```scala
+type Person = (name: String, age: Int)
+val bob: Person
+bob.zip((firstName: String, agee: Int))
+```
+This should report a type error.
+But in the alternative scheme, we'd have `(firstName: String, agee: Int) <: (String, Int)` by subtyping and then
+`(String, Int) -> (name: String, age: Int)` by implicit naming conversion. This is clearly not what we want.
+
+By contrast, in the implemented scheme, we will not convert `(firstName: String, agee: Int)` to `(String, Int)` since a conversion is only attempted if the expected type is a regular tuple, and in our scenario it is a named tuple instead.
+
+My takeaway is that these designs have rather subtle consequences and any alterations would need a full implementation before they can be judged. For instance, the situation with `zip` was a surprise to me, which came up since I first implemented `_.toTuple` as a regular implicit conversion instead of a compiler adaptation.
+
+A possibly simpler design would be to drop all conformance and conversion rules. The problem with this approach is worse usability and problems with smooth migration. Migration will be an issue since right now everything is a regular tuple. If we make it hard to go from there to named tuples, everything will tend to stay a regular tuple and named tuples will be much less used than we would hope for.
+
+
+### Spread Operator
+
+An idea I was thinking of but that I did not include in this proposal highlights another potential problem with subtyping. Consider adding a _spread_ operator `*` for tuples and named tuples. if `x` is a tuple then `f(x*)` is `f` applied to all fields of `x` expanded as individual arguments. Likewise, if `y` is a named tuple, then `f(y*)` is `f` applied to all elements of `y` as named arguments.
+Now, if named tuples would be subtypes of tuples, this would actually be ambiguous since widening `y` in `y*` to a regular tuple would yield a different call. But with the subtyping direction we have, this would work fine.
+
+I believe tuple spread is a potentially useful addition that would fit in well with Scala. But it's not immediately relevant to this proposal, so is left out for now.
+
+
+## Related work
+
+This section should list prior work related to the proposal, notably:
+
+- [Pre-SIP Discussion](https://contributors.scala-lang.org/t/pre-sip-named-tuples/6403)
+
+- [SIP 43 on Pattern Matching with Named Fields](https://github.com/scala/improvement-proposals/pull/44)
+
+- [Experimental Implementation](https://github.com/lampepfl/dotty/pull/19174)
+
+## FAQ
+
+## Appendix A: NamedTuple Definition
+
+Here is the current definition of `NamedTuple`. This is part of the library and therefore subject to future changes and additions.
+
+```scala
+package scala
+import annotation.experimental
+import compiletime.ops.boolean.*
+
+@experimental
+object NamedTuple:
+
+  opaque type AnyNamedTuple = Any
+  opaque type NamedTuple[N <: Tuple, +V <: Tuple] >: V <: AnyNamedTuple = V
```

Comment:
I probably need help from someone else to review this part; this sort of more advanced type-level computation is beyond my expertise

Reply:
As an appendix, it would not need to be reviewed in depth. It just gives additional info that an implementation of `NamedTuple` with a useful API is possible. It's premature to freeze that API, so I believe the NamedTuple API itself should not be part of the SIP, except for things that are explicitly mentioned (like the `NamedTuple` and `From` types).